lms/services/dashboard.py

```diff
@@ -210,7 +211,40 @@ def get_assignment_roster(
                 assignment_id=assignment.id,
                 h_userids=h_userids,
                 # For launch data we always add the "active" column as true for compatibility with the roster query.
-            ).add_columns(True)
+            ).add_columns(true())
+
+        # Always return the results, no matter the source, sorted
+        return roster_last_updated, query.order_by(LMSUser.display_name, LMSUser.id)
+
+    def get_course_roster(
+        self, lms_course: LMSCourse, h_userids: list[str] | None = None
+    ) -> tuple[datetime | None, Select[tuple[LMSUser, bool]]]:
+        rosters_enabled = lms_course.course.application_instance.settings.get(
+            "dashboard", "rosters"
+        )
+        roster_last_updated = self._roster_service.course_roster_last_updated(
+            lms_course
+        )
+        if rosters_enabled and roster_last_updated:
+            # If rostering is enabled and we do have the data, use it
+            query = self._roster_service.get_course_roster(
+                lms_course,
+                role_scope=RoleScope.COURSE,
+                role_type=RoleType.LEARNER,
+                h_userids=h_userids,
+            )
+
+        else:
+            # If we are not going to return data from the roster, don't return the last updated date
+            roster_last_updated = None
```

Comment:
Maybe we can save a query here, and reduce potential errors on forgetting to "unset" `roster_last_updated`, by conditionally fetching it only if rosters are enabled.

```suggestion
        roster_last_updated = self._roster_service.course_roster_last_updated(
            lms_course
        ) if rosters_enabled else None
        if rosters_enabled and roster_last_updated:
            # If rostering is enabled and we do have the data, use it
            query = self._roster_service.get_course_roster(
                lms_course,
                role_scope=RoleScope.COURSE,
                role_type=RoleType.LEARNER,
                h_userids=h_userids,
            )

        else:
```

Reply:
Yes, good suggestion :+1: